lib/php/rendition-factory/src/Transformer/Documentation.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Alchemy\RenditionFactory\Transformer;
+
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+
+class Documentation
+{
+    /** @var Documentation[] */
+    private array $children = [];
+
+    public function __construct(private readonly TreeBuilder $treeBuilder, private readonly string $header = '', private readonly string $footer = '')
+    {
+        $this->children = [];
+    }
+
+    public function addChild(Documentation $child): void
+    {
+        $this->children[] = $child;
+    }
+
+    public function getTreeBuilder(): TreeBuilder
+    {
+        return $this->treeBuilder;
+    }
+
+    public function getChildren(): array
+    {
+        return $this->children;
+    }
+
+    public function getHeader(): string
+    {
+        return $this->header;
+    }
+
+    public function getFooter(): string
+    {
+        return $this->footer;
+    }
+
+    /**
+     * helper to create a base tree for a module, including common options.
+     */
+    public static function createBaseTree(string $name): TreeBuilder
```

Comment:
This should not be related to documentation. Documentation comes over configuration, not the inverse.
In a `TransformerConfigHelper` class?